src/main.rs

```diff
@@ -0,0 +1,272 @@
+    mount_statuses.retain(|ref k, _| {
+        mount_points.iter().position(|i| *i == **k).is_some()
+    });
+
+    for mount_point in mount_points {
+        // Check whether there's a pending test:
+        if let Some(mount_status) = mount_statuses.get_mut(&mount_point) {
+            if mount_status.check_process.is_some() {
+                let child = mount_status.check_process.as_mut().unwrap();
+
+                match child.try_wait() {
+                    Ok(Some(status)) => {
+                        logger
+                            .info(format!(
+                                "Slow check for mount {} exited with {} after {} seconds",
+                                mount_point,
+                                status,
+                                mount_status.last_checked.elapsed().as_secs()
+                            ))
+                            .unwrap_or_else(handle_syslog_error);
+                        ()
+                    }
+                    Ok(None) => {
+                        logger
+                            .warning(format!(
+                                "Slow check for mount {} has not exited after {} seconds",
+                                mount_point,
+                                mount_status.last_checked.elapsed().as_secs()
+                            ))
+                            .unwrap_or_else(handle_syslog_error);
+                        continue;
+                    }
+                    Err(e) => {
+                        logger.err(format!(
+                            "Status update for hung check on mount {} returned an error after {} seconds: {}",
+                            mount_point,
+                            mount_status.last_checked.elapsed().as_secs(),
+                            e
+                        )).unwrap_or_else(handle_syslog_error);
+                        ()
+                    }
+                }
+            }
+        }
+
+        let mount_status = check_mount(&mount_point);
+
+        if mount_status.alive {
+            logger
+                .debug(format!("Mount passed health-check: {}", mount_point))
+                .unwrap_or_else(handle_syslog_error);
+        } else {
+            let msg = format!("Mount failed health-check: {}", mount_point);
+            eprintln!("{}", msg);
+            logger
+                .err(msg)
+                .unwrap_or_else(handle_syslog_error);
+        }
+
+        mount_statuses.insert(mount_point.to_owned(), mount_status);
+    }
+}
+
+fn check_mount(mount_point: &str) -> MountStatus {
+    let mut mount_status = MountStatus {
+        last_checked: Instant::now(),
+        alive: false,
+        check_process: None,
+    };
+
+    let mut child = process::Command::new("/usr/bin/stat")
+        .arg(mount_point)
+        .stdout(process::Stdio::null())
+        .spawn()
+        .unwrap();
+
+    match child.wait_timeout(Duration::from_secs(3)).unwrap() {
```

Comment:
Instead of calling unwrap, you can also match against `Ok(None)` and `Ok(Some(exit_status))`, as well as add a `Err(e) => eprintln!("Error waiting for child:  {:?}", e)`. (Similarly below for rc)